### PR TITLE
MGDAPI-4399 update the docs links and structure

### PR DESCRIPTION
**Pull request by MStokluska**
# Issue link
https://issues.redhat.com/browse/MGDAPI-4399 && https://issues.redhat.com/browse/MGDAPI-4393

# What
Updated docs structure, moved some docs from help repo, fixed broken links

# Verification steps
Locally run make mkdocs/serve and verify that the new layout makes sense and that links are working

### CODE REVIEW

File: docs/products/3scale_guides/tenant_creation_via_cr.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    name: passcredref
+    namespace: 3scale-test
+  systemMasterUrl: 'https://master.apps.pstefans.ooq8.s1.devshift.org'
+  tenantSecretRef:
+   name: tenant1ref
+   namespace: 3scale-test
+  username: tenant1
+```
+
+More information about 3scale tenant can be found in the [3scale Tenant Reference Doc](https://github.com/3scale/3scale-operator/blob/master/doc/tenant-reference.md) . Note: Ensure that the fields are supoorted by checking the [validated_cr_fields](/guides/3scale_capabilities/validated_cr_fields.md) doc.
```
Comment (Fiona-Waters) [inline docs/products/3scale_guides/tenant_creation_via_cr.md:159]: link to validated_cr_fields is broken.

File: docs/products/3scale_guides/create_backend_via_cr.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  namespace: 3scale-test
+spec: 
+  name: "backend1"
+  systemName: "backend"
+  privateBaseURL: "https://api.example.com"
+  providerAccountRef: 
+    name: tenant1ref
+```
+
+More information about 3scale backends can be found in the [3scale Backend Reference Doc](https://github.com/3scale/3scale-operator/blob/master/doc/backend-reference.md) . Note: Ensure that the fields are supoorted by checking the [validated_cr_fields](/guides/3scale_capabilities/validated_cr_fields.md) doc.
```
Comment (Fiona-Waters) [inline docs/products/3scale_guides/create_backend_via_cr.md:81]: link to validated_cr_fields is broken.

File: docs/products/3scale_guides/create_product_via_cr.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+To specify a backend usage in a product add the following under product.spec in the product CR
+```
+backendUsages:
+    <backend_name>:
+      path: /
+```
+
+
+More information about 3scale product can be found in the [3scale Product Reference Doc](https://github.com/3scale/3scale-operator/blob/master/doc/product-reference.md) . Note: Ensure that the fields are supoorted by checking the [validated_cr_fields](/guides/3scale_capabilities/validated_cr_fields.md) doc.
```
Comment (Fiona-Waters) [inline docs/products/3scale_guides/create_product_via_cr.md:85]: link to validated_cr_fields is broken.

File: docs/products/3scale_guides/tenant_creation_via_cr.md (hunk 1 above)
Comment (MStokluska) [reply docs/products/3scale_guides/tenant_creation_via_cr.md:159]: thanks!

File: docs/products/3scale_guides/create_backend_via_cr.md (hunk 2 above)
Comment (MStokluska) [reply docs/products/3scale_guides/create_backend_via_cr.md:81]: thanks!

File: docs/products/3scale_guides/create_product_via_cr.md (hunk 3 above)
Comment (MStokluska) [reply docs/products/3scale_guides/create_product_via_cr.md:85]: thanks!

**Comment by Fiona-Waters** [comment]
Looks good, no broken links.
/lgtm

File: docs/index.md [hunk 4]
```diff
@@ -6,11 +6,11 @@ An Integreatly Operator can be installed using three different flavours: `manage
 
 To switch between the three you can use export the `INSTALLATION_TYPE` env or use it in conjunction with any of the make commands referenced in this README
 
-### Installed products
+## Installed products
 
 The operator installs the following products:
 
-### managed
+## managed
```
Comment (Boomatang) [inline docs/index.md:13]: The heads managed, managed-api and multitenant-managed-api make more sense to be nested under Installed products. This to have being at fault in the original document.
```suggestion
### managed
```

File: docs/installation_guides/installation_general_info.md [hunk 5]
```diff
@@ -1,4 +1,7 @@
-# Installing RHMI Using OCM
+# Overview
```
Comment (Boomatang) [inline docs/installation_guides/installation_general_info.md:1]: I know you only added the Overview section. But I am question what is the need for this document. The getting started guide list cluster as a prerequisite. That requisite does need update as it lists 4.6 as the version but that is a different matter.

It would be of my opinion that this document can be completely removed.

File: docs/installation_guides/validation_of_installation.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
+# Validate installation
```
Comment (Boomatang) [inline docs/installation_guides/validation_of_installation.md:1]: Should this be a separate document? in the 3scale_guides.md this information is listed as a section. Should this be added to other documents?

This is a general question and I am looking for your opinion.

File: docs/products/custom_domain.md [hunk 7]
```diff
@@ -13,7 +13,7 @@ Required Items
 * CNAME edit access for a domain name
 
 
-### Domain Certificate
+## Domain Certificate
```
Comment (Boomatang) [inline docs/products/custom_domain.md:16]: This part of the cluster configuration. It should be nested as such.
```suggestion
### Domain Certificate
```

File: docs/products/custom_domain.md [hunk 8]
```diff
@@ -67,7 +67,7 @@ The endpoint URL will need to be added to a CNAME record for the domain in quest
 Team leads should be able to help get access to the required resources.
 This guide does not cover the steps required to configure the CNAME record in the DNS configuration.
 
-## Configuring the operator installation.
+<h5><b>Configuring the operator installation</b></h5>
```
Comment (Boomatang) [inline docs/products/custom_domain.md:70]: Why this change? Adding the h5 puts this item under domain certificate which it does not belong to. Why would you want this section not linkable?

File: docs/products/products_general_info.md [hunk 9]
```diff
@@ -0,0 +1,3 @@
+# Products guides overview
```
Comment (Boomatang) [inline docs/products/products_general_info.md:1]: What is need for such a short document?

File: docs/uninstallation/uninstallation_general_info.md [hunk 10]
```diff
@@ -0,0 +1,2 @@
+# Uninstalling RHOAM
```
Comment (Boomatang) [inline docs/uninstallation/uninstallation_general_info.md:1]: Same question as before. What is the need for such a short document?

File: mkdocs.yml [hunk 11]
```diff
@@ -5,3 +5,32 @@ theme:
...
+    - Tests:
+      - Unit testing: tests/unit_tests.md
+      - End to End (e2e): tests/e2e.md
+      - Product tests: tests/product_tests.md
+      - Scorecard tests: tests/scorecard_tests.md
+    - Uninstallation:
+      - Uninstallation guides: uninstallation/uninstallation_general_info.md
+      - Local and OLM: uninstallation/local_and_olm.md
+      - Addon flow: uninstallation/addon_flow.md
+      - ROSA: uninstallation/rosa.md
```
Comment (Boomatang) [inline mkdocs.yml:34]: There is a couple points. 
Firstly with the change as it is there is a number of page where the page title does not match the nav item name. From what I have seen there is no reason why it needs to be done in this fashion.

Secondly why add this at all? Does the order matter? The name of the folders act as the section headers. If the section headers need to be updated changing the folder names would also work.  If possible I would like to avoid adding the nav section here. With out it any doc added to the docs folder is auto added to the site. It is less of maintenance overhead. If it is for ordering I understand that too.

Recap.
Some document titles do not match their navigation titles. This needs updating.
Why add the nav section? Is it to force the order sections are listed?

**Review by Boomatang — CHANGES_REQUESTED** [review]
Some comments are asking for clarification on the changes. But there is issue that need to be addressed.

File: mkdocs.yml (hunk 11 above)
Comment (MStokluska) [reply mkdocs.yml:34]: Good point on naming, going to update.
As for nav, yes, without nav the order is alphabetical while to me docs should have a clear structure, for example, I do want to cover stuff like installation or pre-requisites before tests or uninstallation. We need nav configured IMO

Comment (MStokluska) [reply mkdocs.yml:34]: Actually, looking at it now, which names mismatch in your opinion?

File: docs/index.md (hunk 4 above)
Comment (MStokluska) [reply docs/index.md:13]: Good point, ty

File: docs/installation_guides/installation_general_info.md (hunk 5 above)
Comment (MStokluska) [reply docs/installation_guides/installation_general_info.md:1]: Agree, removed

File: docs/installation_guides/validation_of_installation.md (hunk 6 above)
Comment (MStokluska) [reply docs/installation_guides/validation_of_installation.md:1]: To me, validation of installation should be a separate page since we have split the installation with OLM vs Local, so it makes me think it's logical to have the verification as separate step. 
An alternative to this is to duplicate the verification and list it both in olm and local, however, this brings duplication which I think we can avoid.

File: docs/products/custom_domain.md (hunk 7 above)
Comment (MStokluska) [reply docs/products/custom_domain.md:16]: Good point, ty

File: docs/products/custom_domain.md (hunk 8 above)
Comment (MStokluska) [reply docs/products/custom_domain.md:70]: As agreed, I've moved away from html addition, this is a miss.

File: docs/products/products_general_info.md (hunk 9 above)
Comment (MStokluska) [reply docs/products/products_general_info.md:1]: All the "overviews" came from the initial implementation of the html. Going to remove, thanks @Boomatang

File: docs/uninstallation/uninstallation_general_info.md (hunk 10 above)
Comment (MStokluska) [reply docs/uninstallation/uninstallation_general_info.md:1]: same as above, ty

File: docs/installation_guides/validation_of_installation.md (hunk 6 above)
Comment (Boomatang) [reply docs/installation_guides/validation_of_installation.md:1]: Ya I am happy with that. I will say the more I look at the command for checking if the installation is complete, the more I think that could be convert into a one liner for all types. That update is outside of this work.

File: mkdocs.yml (hunk 11 above)
Comment (MStokluska) [reply mkdocs.yml]: ah, nvm, I read your initial comment as the name of the files are different to the nav items...
going to update now, thanks

Comment (MStokluska) [reply mkdocs.yml]: @Boomatang pushed changes now, thanks

**Review by Boomatang — APPROVED** [review]
/lgtm
